_pytest/logging.py

```diff
@@ -337,8 +342,14 @@ def __init__(self, config):
         """
         self._config = config
 
+        auto_enable_log_stream = self._config.getoption('--log-cli-level')
+        if auto_enable_log_stream is not None:
+            # Explicitely enable log_cli is --log-cli-level was passed on the CLI
+            # It won't be automatically enabled is set on the ini file.
+            config._inicache['log_cli'] = config._parser._inidict['log_cli'] = True
```

Comment:
I don't think we should be changing the `config._*` attributes, I prefer we isolate this predicate into a method and use that instead, for example:

```
def _stream_logs(self):
    return self._config.getoption('--log-cli-level') is not None or self._config.getini('log_cli')
```